core/common/src/main/java/alluxio/ParameterizedPropertyKey.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.exception.ExceptionMessage;
+
+import com.google.common.base.Objects;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A set of parameterized property keys that can generate the property name given parameters.
+ * E.g., * {@code ParameterizedPropertyKey.MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS.format(0)}
+ */
+@ThreadSafe
+public enum ParameterizedPropertyKey {
+  MASTER_MOUNT_TABLE_ENTRY_ALLUXIO("alluxio.master.mount.table.%s.alluxio",
+      "alluxio\\.master\\.mount\\.table.(\\w+)\\.alluxio"),
+  MASTER_MOUNT_TABLE_ENTRY_OPTION("alluxio.master.mount.table.%s.option",
+      "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.option"),
+  MASTER_MOUNT_TABLE_ENTRY_OPTION_PROPERTY("alluxio.master.mount.table.%s.option.%s",
+      "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.option(\\.\\w+)++"),
+  MASTER_MOUNT_TABLE_ENTRY_READONLY("alluxio.master.mount.table.%s.readonly",
+      "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.readonly"),
+  MASTER_MOUNT_TABLE_ENTRY_SHARED("alluxio.master.mount.table.%s.shared",
+      "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.shared"),
+  MASTER_MOUNT_TABLE_ENTRY_UFS("alluxio.master.mount.table.%s.ufs",
+      "alluxio\\.master\\.mount\\.table\\.(\\w+)\\.ufs"),
+  MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS("alluxio.master.tieredstore.global.level%d.alias",
+      "alluxio\\.master\\.tieredstore\\.global\\.level(\\d+)\\.alias"),
+  WORKER_TIERED_STORE_LEVEL_ALIAS("alluxio.worker.tieredstore.level%d.alias",
+      "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.alias"),
+  WORKER_TIERED_STORE_LEVEL_DIRS_PATH("alluxio.worker.tieredstore.level%d.dirs.path",
+      "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.dirs\\.path"),
+  WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA("alluxio.worker.tieredstore.level%d.dirs.quota",
+      "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.dirs\\.quota"),
+  WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO("alluxio.worker.tieredstore.level%d.reserved.ratio",
+      "alluxio\\.worker\\.tieredstore\\.level(\\d+)\\.reserved\\.ratio"),
+  ;
+
+  private final String mFormat;
+  private final Pattern mPattern;
+
+  /**
+   * @param input string of property key
+   * @return whether the input is a valid property name
+   */
+  public static boolean isValid(String input) {
+    for (ParameterizedPropertyKey key : values()) {
+      Matcher matcher = key.mPattern.matcher(input);
+      if (matcher.matches()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Parses a string and return its corresponding {@link PropertyKey}, throwing exception if no such
+   * a property can be found.
+   *
+   * @param input string of property key
+   * @return corresponding property
+   */
+  public static PropertyKey fromString(String input) {
```

Comment:
I think this method should be called `getPropertyKey()` or `matchPropertyKey()` ... I would `T.fromString` to return object of type `T`